tests/src/Functional/DeveloperSyncTest.php

```diff
@@ -0,0 +1,216 @@
+use Drupal\apigee_edge\Entity\DeveloperInterface;
+use Drupal\user\Entity\User;
+use Drupal\user\UserInterface;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Output\BufferedOutput;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Developer synchronization test.
+ *
+ * @group apigee_edge9
+ */
+class DeveloperSyncTest extends ApigeeEdgeFunctionalTestBase {
+
+  public static $modules = [
+    'block',
+  ];
+
+  /**
+   * Random property prefix.
+   *
+   * @var string
+   */
+  protected $prefix;
+
+  /**
+   * Email filter.
+   *
+   * @var string
+   */
+  protected $filter;
+
+  /**
+   * Array of Apigee Edge developers.
+   *
+   * @var \Drupal\apigee_edge\Entity\DeveloperInterface[]
+   */
+  protected $edgeDevelopers = [];
+
+  /**
+   * Array of Drupal users.
+   *
+   * @var \Drupal\user\UserInterface[]
+   */
+  protected $drupalUsers = [];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->prefix = $this->randomMachineName();
+    $escaped_prefix = preg_quote($this->prefix);
+    $this->filter = "/^{$escaped_prefix}\.[a-zA-Z0-9]*@example\.com$/";
+    $this->container->get('config.factory')->getEditable('apigee_edge.sync')->set('filter', $this->filter)->save();
+
+    // Create developers on Apigee Edge.
+    for ($i = 0; $i < 1; $i++) {
+      $this->edgeDevelopers[$i] = Developer::create([
+        'email' => "{$this->prefix}.{$this->randomMachineName()}@example.com",
+        'userName' => $this->randomMachineName(),
+        'firstName' => $this->randomMachineName(),
+        'lastName' => $this->randomMachineName(),
+      ]);
+      $this->edgeDevelopers[$i]->save();
+    }
+
+    // Create users in Drupal.
+    _apigee_edge_set_sync_in_progress(TRUE);
+    for ($i = 0; $i < 1; $i++) {
+      $this->drupalUsers[$i] = $this->createAccount([], TRUE, $this->prefix, FALSE);
+    }
+    _apigee_edge_set_sync_in_progress(FALSE);
+
+    $this->drupalLogin($this->rootUser);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function tearDown() {
+    $remote_ids = array_map(function ($record): string {
+      return $record['email'];
+    }, $this->edgeDevelopers);
+    $drupal_emails = array_map(function (UserInterface $user): string {
+      return $user->getEmail();
+    }, $this->drupalUsers);
+    $ids = array_merge($remote_ids, $drupal_emails);
+    foreach ($ids as $id) {
+      Developer::load($id)->delete();
+    }
+    parent::tearDown();
+  }
+
+  /**
+   * Verifies that the Drupal users and the Edge developers are synchronized.
+   */
+  protected function verify() {
+    $all_users = [];
+    /** @var \Drupal\user\UserInterface $account */
+    foreach (User::loadMultiple() as $account) {
+      $email = $account->getEmail();
+      if ($email && $email !== 'admin@example.com') {
+        $this->assertTrue($this->filter ? (bool) preg_match($this->filter, $email) : TRUE, "Email ({$email}) is filtered properly.");
```

Comment:
Don't we need a proper error message to be display if this assert fails? (Second parameter of method).